python/taichi/lang/matrix.py

```diff
@@ -363,13 +363,22 @@ def transposed(a):
     def T(self):
         return self.transpose()
 
+    @deprecated('ti.transpose(a)', 'a.transpose()')
     def transpose(a):
         ret = Matrix(a.m, a.n, empty=True)
```

Comment:
How about just return `a.transpose()`, so that we don't repeat ourselves? Same for `determinant`?